### PR TITLE
Issue #21840: Remove distributedSecurityCache-1.0 feature.

### LINKED ISSUE #21840: SEC 380: Remove distributedSecurityCache-1.0 feature
The javax.cache packages are exposed as API in this feature. Expose them to the jcache implementation library internally using the SpiLibrary class.

### PR DESCRIPTION
**Pull request by jvanhill**
fixes #21840

### CODE REVIEW

**Comment by posmith** [comment]
L2 message review completed

**Comment by jvanhill** [comment]
#build
#libby

**Comment by jvanhill** [comment]
Respun build as there was an issue in the build.

https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_TBj3sAz8Ee2vacLbpVXjpQ

File: dev/com.ibm.ws.security.authentication.builtin/resources/com/ibm/ws/security/authentication/internal/resources/AuthenticationMessages.nlsprops [hunk 1]
```diff
@@ -83,3 +83,7 @@ JAAS_LOGIN_COLLECTIVE_PLUGIN_UNAVAILABLE.useraction=No action is required.
 JCACHE_AUTH_CACHE_CLEARED_ALL_ENTRIES=CWWKS1125I: The authentication cache cleared all entries in the {0} JCache cache. 
 JCACHE_AUTH_CACHE_CLEARED_ALL_ENTRIES.explanation=The authentication cache clears all entries in the backing JCache cache when security or user registry configuration is modified, or when the DeleteAuthCache MBean is called. 
 JCACHE_AUTH_CACHE_CLEARED_ALL_ENTRIES.useraction=No action is required.
+
+JCACHE_AUTH_CACHE_SERIALIZATION_FAILED=CWWKS1126W: Insertion of entry for the {0} key into the {1} JCache failed due to a serialization error. The entry will be inserted into the in-memory cache instead. This warning will not be emitted again. The exception is: {1}
+JCACHE_AUTH_CACHE_SERIALIZATION_FAILED.explanation=The contents of the entry could not be serialized so they could not be stored in the JCache cache.
```
Comment (helyarp) [inline dev/com.ibm.ws.security.authentication.builtin/resources/com/ibm/ws/security/authentication/internal/resources/AuthenticationMessages.nlsprops]: Change to remove the "could":

The entry contents were not serialized so they might not be stored in the JCache cache.

File: dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops [hunk 2]
```diff
@@ -48,9 +48,9 @@ CWLJC0005_MANAGER_LOADED=CWLJC0005I: The {0} JCache CacheManager took {1} ms to
 CWLJC0005_MANAGER_LOADED.explanation=The JCache CacheManager was successfully loaded.
 CWLJC0005_MANAGER_LOADED.useraction=No user action required.
 
-#
-# CWLJC0006 IS AVAILABLE.
-#
+CWLJC0006_MULTI_REF_LIB=CWLJC0006E: The {0} caching provider referenced the {1} library from both jCacheLibaryRef and commonLibraryRef. This is not supported. 
+CWLJC0006_MULTI_REF_LIB.explanation=Referencing the same library from the commonLibraryRef and the jCacheLibraryRef can lead to undefined behavior due to classloading issues. 
+CWLJC0006_MULTI_REF_LIB.useraction=The commonLibraryRef may not reference the same library as the jCacheLibraryRef.
```
Comment (helyarp) [inline dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops]: Change "may" to "might"

**Review by helyarp — COMMENTED** [review]
Suggested changes

File: dev/com.ibm.ws.security.authentication.builtin/resources/com/ibm/ws/security/authentication/internal/resources/AuthenticationMessages.nlsprops (hunk 1 above)
Comment (jvanhill) [reply dev/com.ibm.ws.security.authentication.builtin/resources/com/ibm/ws/security/authentication/internal/resources/AuthenticationMessages.nlsprops]: Changed to:

`The entry contents were not serialized so they were not stored in the JCache cache.`

File: dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops (hunk 2 above)
Comment (jvanhill) [reply dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops]: The statement is an absolute... The commonLibraryRef and jCacheLibraryRef cannot have the same library referenced.

Changed to:

`The commonLibraryRef cannot reference the same library as the jCacheLibraryRef.`

**Comment by jvanhill** [comment]
#build
#libby

**Comment by jvanhill** [comment]
#build
#libby

**Review by dazavala — APPROVED** [review]
Looks good.  Config changes and runtime agree with previous discussion with Alasdair and Jared.  Great!

**Review by helyarp — COMMENTED** [review]
Thanks for making the changes.

**Review by kristip17 — APPROVED** [review]

**Comment by jvanhill** [comment]
#build
#libby

**Comment by jvanhill** [comment]
The last build passed 100% except for one FAT child build that appears stuck. Respinning.

#build

**Comment by jvanhill** [comment]
The failures in the latest build were pre-existing issues.

**Pull request merged**